src/berry_mill/kiwrap.py

```diff
@@ -107,9 +120,17 @@ def _get_repokeys(self, reponame: str, repodata: Dict[str, str]) -> str:
         else:
             s_url = urljoin(f"{url.scheme}://{url.netloc}/{url.path}/Release.key", "")
             response = requests.get(s_url, allow_redirects=True)
-            with open(g_path, 'xb') as f_rel:
-                f_rel.write(response.content)
-            response.close()
+            # check reponse OK
+            if response.status_code == HTTPStatus.OK:
+                with open(g_path, 'xb') as f_rel:
+                    f_rel.write(response.content)
+                response.close()
+            else:
+                raise SystemExit(f"Wrong url defined for repo {reponame}")
```

Comment:
I wouldn't exit here.... 
There might be a repository which exists but does not have a file that is called "Release.key" and rather something else or no such file entirely.

Displaying a Warning to the User here should be enough.
If you return an empty g_path here, berrymill should already automatically ask the user for a valid key.
By now the user should know that something went wrong or that they need to fetch a key themselves